src/pages/edge/adobe-journey-optimizer/code-based/tutorial.md

```diff
@@ -0,0 +1,114 @@
+
+## Pre-requisites
+
+[Integrate and register Messaging extension](../../../index.md#implement-extension-in-mobile-app) in your app.
+
+## Fetch and cache the code-based content
+
+To fetch the content for the surfaces configured in Adobe Journey Optimizer campaigns, call the [updatePropositionsForSurfaces](../api-reference.md#updatepropositionsforsurfaces) API . It is encouraged to batch requesting multiple [Surface](../public-classes/surface.md) URIs in a single API call when possible. The returned code-based experiences are cached in-memory by the Messaging extension and persists through the lifecycle of the app (i.e as long as the app is running). An example of the call is shown below.
+
+<TabsBlock orientation="horizontal" slots="heading, content" repeat="2"/>
+
+Android
+
+<Tabs query="platform=android&function=updatepropositionsforsurfaces"/>
+
+iOS
+
+<Tabs query="platform=ios&function=updatepropositionsforsurfaces"/>
+
+## Retrieve cached propositions
+
+To retrieve the previously fetched content from the in-memory cache, call [getPropositionsForSurfaces](../api-reference.md#getpropositionsforsurfaces)  API with a list of required surface URIs and a completion handler. The completion handler will be invoked with a list of [Proposition](../public-classes/proposition.md) objects corresponding to the requested surfaces, or AEPError object if an error occurs. If a requested surface was not previously cached prior to calling `getPropositionsForSurfaces` (using the `updatePropositionsForSurfaces` API), no propositions will be returned for that surface. The below example shows how to retrieve the content for the previously cached surfaces.
+
+<TabsBlock orientation="horizontal" slots="heading, content" repeat="2"/>
+
+Android
+
+<Tabs query="platform=android&function=getpropositionsforsurfaces"/>
+
+iOS
+
+<Tabs query="platform=ios&function=getpropositionsforsurfaces"/>
+
+## Using the retrieved propositions
+
+The [Proposition](../public-classes/proposition.md) object returned in the completion handler encapsulates the content specified for the corresponding surface, as well as information needed for tracking interactions with the content. Multiple Proposition  objects can be returned for a single surface based on the number of campaigns configured for it in Adobe Journey Optimizer. Each Proposition  object in turn can contain multiple items, represented by the [PropositionItem](../public-classes/proposition-item.md) class, based on how the campaign's content is defined. To access the content, iterate through the list of PropositionItem  present in the returned list of Propositions. The SchemaType of the PropositionItem indicates the type of content it contains and can be used to determine how to render or interpret the returned content. The PropositionItem  class contains helper functions to access the different types of supported content.
```

Comment:
Double space between "Proposition" and "objects" throughout here.

And also one between the "PropositionItem" and "present".